File: ch13/dotnet/change-feed/BlobChangeFeedExample.java
// Java example: Reading Azure Blob Storage Change Feed
// Requires Maven dependency: com.azure:azure-storage-blob-changefeed

import com.azure.storage.blob.BlobServiceClient;
import com.azure.storage.blob.BlobServiceClientBuilder;
import com.azure.storage.blob.changefeed.BlobChangefeedClient;
import com.azure.storage.blob.changefeed.BlobChangefeedClientBuilder;
import com.azure.storage.blob.changefeed.models.BlobChangefeedEvent;

public class BlobChangeFeedExample {
    public static void main(String[] args) {
        // Replace with your Blob service endpoint and credentials
        String connectionString = "<your_connection_string>";

        // Create a BlobServiceClient
        BlobServiceClient serviceClient = new BlobServiceClientBuilder()
                .connectionString(connectionString)
                .buildClient();

        // Create a BlobChangefeedClient
        BlobChangefeedClient changefeedClient = new BlobChangefeedClientBuilder(serviceClient).buildClient();

        // Read change feed events
        for (BlobChangefeedEvent event : changefeedClient.getEvents()) {
            System.out.println("Event Type: " + event.getEventType());
            System.out.println("Subject: " + event.getSubject());
            System.out.println("Event Time: " + event.getEventTime());
            System.out.println("---");
        }
    }
}

File: ch13/dotnet/change-feed/blob_change_feed_example.py
# Python example: Reading Azure Blob Storage Change Feed
# Requires: pip install azure-storage-blob-changefeed

from azure.storage.blob import BlobServiceClient
from azure.storage.blob.changefeed import BlobChangeFeedClient

# Replace with your Blob service connection string
connection_string = "<your_connection_string>"

# Create a BlobServiceClient
service_client = BlobServiceClient.from_connection_string(connection_string)

# Create a BlobChangeFeedClient
change_feed_client = BlobChangeFeedClient(service_client)

# Read change feed events
for event in change_feed_client.list_changes():
    print(f"Event Type: {event.event_type}")
    print(f"Subject: {event.subject}")
    print(f"Event Time: {event.event_time}")
    print("---")
